frontend-ssr/components/cli_git/GitCli_CC.tsx

```diff
@@ -0,0 +1,135 @@
+import {uuid} from '../../services/Utils';
+
+import styles from './GitCli.module.scss';
+
+interface Props {
+  commands: RegExp[] | string[],
+  successMessages: string[],
+}
+
+interface State {
+  currentIndex: number,
+  messages: Message[],
+  btnMessage: string,
+}
+
+interface Message {
+  text: string,
+  type: MessageType,
+  time: string,
+  id: string,
+}
+
+enum MessageType {
+  COMMAND = 'command',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+export default class MatchCommandClassComponent extends React.Component<Props, State, Message> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentIndex: 0,
+      messages: [],
+      btnMessage:  'Next',
+    };
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    const formEl = event.target;
+    const gitCommand = event.target.gitCommand.value;
+    const { commands, successMessages } = this.props;
+    const {currentIndex} = this.state;
+
+    let isMatch = false;
+
+    if (commands[currentIndex] instanceof RegExp) {
+      isMatch = (commands[currentIndex] as RegExp).test(gitCommand);
+    } else {
+      isMatch = commands[currentIndex] === gitCommand.trim().toLowerCase();
+    }
+
+    const date = new Date();
+
+    const commandMessage: Message = {
+      text: gitCommand,
+      type: MessageType.COMMAND,
+      time: date.toLocaleTimeString(),
+      id: uuid()
+    };
+  
+    const successMessage : Message = {
+      text: successMessages[currentIndex],
+      type: MessageType.SUCCESS,
+      time: date.toLocaleTimeString(),
+      id: uuid()
+    };
+
+    const errorMessage: Message = {
+      text: 'This is not a valid command',
+      type: MessageType.ERROR,
+      time: date.toLocaleTimeString(),
+      id: uuid()
+    };
+
+    if (!isMatch) {
+      this.setState(({ messages }) => ({
+        messages: [...messages, commandMessage, errorMessage],
+        btnMessage: 'Next',
+      }));
+    } else if (currentIndex === commands.length - 1) {
+      this.setState(({ messages }) => ({
+        messages: [...messages, commandMessage, successMessage],
+        currentIndex: 0,
+        btnMessage: 'Done',
+      }));
+    } else {
+      this.setState(({ messages }) => ({
+        messages: [...messages, commandMessage, successMessage],
+        currentIndex: currentIndex + 1,
+      }));
+    };
+    formEl.reset();
+  };
+
+  render() {
+    const { messages, btnMessage } = this.state;
+
+    return (
+      <>
+        <div className={styles.commands}>
+          <div
+            className={styles.commands__cli}
+          >
+            {messages.map((message) => (
+              <p key={message.id} className={styles[message.type]}>
+                {message.text}
+                {' '}
+                {message.time}
+              </p>
+            ))}
+          </div>
+          <form
+            className={styles.form}
+            onSubmit={this.handleSubmit}
+          >
+            <input
```

Comment:
Let's add `spellCheck="false"` as an attribute. Since this commands may not be valid english, we don't want the red lines underneath.